x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/QueryConfig.java

```diff
@@ -98,6 +102,20 @@ public static QueryConfig fromXContent(final XContentParser parser, boolean leni
         return new QueryConfig(source, query);
     }
 
+    private static QueryBuilder queryFromXContent(
+        Map<String, Object> source,
+        NamedXContentRegistry namedXContentRegistry,
+        DeprecationHandler deprecationHandler
+    ) throws IOException {
+        QueryBuilder query = null;
```

Comment:
Is this local variable needed at all?

Reply:
yes, for lenient parsing we only log the exception and query remains `null`. I could also have 2 return's, but I think 1 exit is better than 2.